File: NeuronNetworks/Perseptron/Layers/HiddenLayer.cs
using NeuronNetworks.Perseptron.Interfaces;

namespace NeuronNetworks.Perseptron.Layers
{
    public class HiddenLayer : OutputLayer, IHiddenLayer
    {
        public HiddenLayer(int neuronsCount, ILayer previousLayer, int index) : base(neuronsCount, previousLayer, index)
        {

        }

        public ILayer NextLayer { get; set; }

        public override double[] Handle(double[] inputs)
        {
            var neuronsResult = base.Handle(inputs);

            return NextLayer.Handle(neuronsResult);
        }
    }
}

File: JavaScript (TypeScript)/src/neuron-network/perseptron/Layers/hidden-layer.ts
import { IHiddenLayer } from '../abstract';
import { OutputLayer } from './output-layer';
import { ILayer } from '../../abstract';

export class HiddenLayer extends OutputLayer implements IHiddenLayer {
    nextLayer: ILayer;

    /**
     * HiddenLayer
     */
    constructor(neuronsCount: number, previousLayer: ILayer, index: number) {
        super(neuronsCount, previousLayer, index);
    }
}
